receiver/receivercreator/factory.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receivercreator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector/config/configerror"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/receiver"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+// This file implements factory for receiver_creator. A receiver_creator can create other receivers at runtime.
+
+const (
+	typeStr = "receiver_creator"
+)
+
+// Factory is the factory for receiver_creator.
+type Factory struct {
+}
+
+var _ receiver.Factory = (*Factory)(nil)
```

Comment:
`receiver.Factory` is renamed to `component.ReceiverFactoryOld`

Reply:
Will update to use new interfaces.

Reply:
Oh looks like I can't update yet because of data package still being internal.

Reply:
Yes, please use the Old interface for now. Once Bogdan is done with new structures we will make them public.